### PR TITLE
Fix warning: overlapping access to 'self'

### PR DESCRIPTION
**Pull request by ianpartridge**
> Warning: Overlapping accesses to 'self', but modification requires exclusive access; consider copying to a local variable

There are more of these in the codebase, but here's a fix for a few.

### CODE REVIEW

File: Foundation/NSCalendar.swift [hunk 1]
```diff
@@ -583,8 +583,9 @@ open class NSCalendar : NSObject, NSCopying, NSSecureCoding {
         var at: CFAbsoluteTime = date.timeIntervalSinceReferenceDate
         
         let res: Bool = withUnsafeMutablePointer(to: &at) { t in
+            let copy = vector
             return vector.withUnsafeMutableBufferPointer { (vectorBuffer: inout UnsafeMutableBufferPointer<Int32>) in
```
Comment (spevans) [inline Foundation/NSCalendar.swift]: Would `let count = Int32(vector.count)` be better to reduce the size of the copy?

File: TestFoundation/TestNSData.swift [hunk 2]
```diff
@@ -924,13 +924,14 @@ extension TestNSData {
...
         // Get the underlying data for hello2
+        let copy = mutatingHello
         mutatingHello.withUnsafeMutableBytes { (bytes : UnsafeMutablePointer<UInt8>) in
             XCTAssertEqual(bytes.pointee, 0x68, "First byte should be 0x68")
             
             // Mutate it
             bytes.pointee = 0x67
             XCTAssertEqual(bytes.pointee, 0x67, "First byte should be 0x67")
-            XCTAssertEqual(mutatingHello[0], 0x67, "First byte accessed via other method should still be 0x67")
+            XCTAssertEqual(copy[0], 0x67, "First byte accessed via other method should still be 0x67")
```
Comment (spevans) [inline TestFoundation/TestNSData.swift]: Isn't this copy making the test redundant?
Would moving the `XCTAssertEqual(mutatingHello[0] ... ` outside of the closure give the same affect?

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by spevans** [comment]
Those fixes look good.

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**